### PR TITLE
dev/core#5403 fix crash installing an extension that provides an EntityBridge

### PR DESCRIPTION
**Pull request by ufundo**
Overview
----------------------------------------
Fix for https://lab.civicrm.org/dev/core/-/issues/5403 - alternative to https://github.com/civicrm/civicrm-core/pull/30912 

Before
---------------------------------------
- expired UserJobs are deleted early in a cache clear triggered using `rebuildMenuAndCaches`
- installing a new Standalone site crashes at the point it tries to install `standaloneusers`
- ~~_to be confirmed:_ installing any extension that provides an entity that uses the EntityBridge trait crashes~~
- installing [this demo extension](https://lab.civicrm.org/ufundo/wobbly_bridge) causes a similar crash on D7

After
--------------------------------------
- UserJobs are deleted near the end of the cache clear process
- no crashes

Technical details
--------------------------------------
See https://lab.civicrm.org/dev/core/-/issues/5403#note_169033 for my theory of the crime :dagger:

### CODE REVIEW

**Comment by ufundo** [comment]
As compared to https://github.com/civicrm/civicrm-core/pull/30912 this seems a less complete fix.

However it has the advantage of actually fixing the issue.

**Comment by ufundo** [comment]
Adding extended tests to check Standalone install and potentially flush out any other bootstrap type edge cases

**Comment by ufundo** [comment]
Pretty sure all those extended test fails look familiar from the baseline. (@totten - is there any way to automatically cross-reference tests fails on Jenkins?)

So even if not the final fix, I think this is better out than in.

**Comment by totten** [comment]
@ufundo 

* Wow, this is really good investigation.
* I'll look through the test comparison to be sure about the pre-existing failures.
* IMHO, it makes a lot of sense to move back calls for `Civi\Api4\UserJob` (*which relies on more high-level services*) to somewhere after `CRM_Core_DAO_AllCoreTables::flush()` and `CRM_Utils_System::flushCache()` (*which clear some low-level services*).
* I share your sense that the lifecycle needs some kind of clarification. FWIW:
    * The whole callgraph around reset/cleanup feels precarious.
    * Within the current framework... `cleanupCaches()` *feels* generally more low-level -- and [this area](https://github.com/civicrm/civicrm-core/blob/c52072209f4184d4074a0eb2f68cfc3793677dba/CRM/Core/Invoke.php#L403-L417) *feels* a bit more high-level. I would guess `rebuildMenuAndCaches()` to be a bit more appropriate for `UserJob::delete()`.
* The topic is a bit pressing b/c it obstructs all other work on standalone. So +1 for targeted fix for now.

(*Let me finish comparing the test results -- and then we'll merge, unless there's some intervening objection.*)

**Comment by totten** [comment]
Yup, all the extended failures are pre-existing. This does, in fact, fix the failures around installing Standalone. 🎉

**Comment by ufundo** [comment]
> cleanupCaches() feels generally more low-level -- and this area feels a bit more high-level. I would guess rebuildMenuAndCaches() to be a bit more appropriate for UserJob::delete().

As far as I could see `rebuildMenuAndCaches` is the only caller for `cleanupCaches`? so maybe they could just be consolidated. 

But it worries me that as it stands there's no easy way to put a health warning saying "dont use api4 between _here_ and _here_". It would be good for the danger zone to basically be within a single, well documented function call, right?

File: CRM/Core/Config.php [hunk 1]
```diff
@@ -295,6 +294,10 @@ public function cleanupCaches($sessionReset = FALSE) {
     CRM_Core_DAO_AllCoreTables::flush();
     CRM_Utils_System::flushCache();
 
+    // note this used to be earlier, but was crashing because of api4 instability
+    // during extension install
+    UserJob::delete(FALSE)->addWhere('expires_date', '<', 'now')->execute();
```
Comment (colemanw) [inline CRM/Core/Config.php:299]: @ufundo IMO we are just asking for trouble by calling the api from within super low-level functions like this one. I would suggest calling the BAO directly for this instead.
